File: 2021/src/test/java/de/jfriemel/aoc/days/Day15Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day15Test {

    private static final String INPUT_STRING = """
            1163751742
            1381373672
            2136511328
            3694931569
            7463417111
            1319128137
            1359912421
            3125421639
            1293138521
            2311944581""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day15().part1(INPUT);
        assertEquals("40", result);
    }

    @Test
    public void testPart2() {
        String result = new Day15().part2(INPUT);
        assertEquals("315", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day15Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day15Test {
    private val input = "rn=1,cm-,qp=3,cm=2,qp-,pc=4,ot=9,ab=5,pc-,pc=6,ot=7".lines()

    @Test
    fun testPart1() {
        val result = Day15.part1(input)
        Assertions.assertEquals("1320", result)
    }

    @Test
    fun testPart2() {
        val result = Day15.part2(input)
        Assertions.assertEquals("145", result)
    }
}
